addons/sale/models/sale_order.py

```diff
@@ -597,6 +602,14 @@ def _onchange_commitment_date(self):
                 }
             }
 
+    @api.depends('company_id', 'partner_credit', 'partner_credit_limit', 'amount_total')
+    def _compute_show_partner_credit_warning(self):
+        for order in self:
+            order.with_company(order.company_id)
+            order.partner_updated_credit = order.partner_credit + order.amount_total
```

Comment:
`order.amount_total` is expressed in `order.currency_id` whereas `order.partner_credit` & `order.partner_credit_limit` are expressed in the company currency (I suppose).
A currency conversion is probably needed.
Also, why don't you consider the SO state directly in the compute instead of combining both checks in the view ?
IMHO, it would be clearer to know when and why the warning is shown ;).

Reply:
Good points, thanks for the heads up :slightly_smiling_face: